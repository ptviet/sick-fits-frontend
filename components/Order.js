import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Head from 'next/head';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { format } from 'date-fns';
import formatMoney from '../lib/formatMoney';
import ErrorMessage from './ErrorMessage';
import OrderStyles from './styles/OrderStyles';
import OrderItem from './OrderItem';

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(id: $id) {
      id
      charge
      total
      createdAt
      user {
        id
      }
      items {
        id
        title
        description
        price
        image
        largeImage
        quantity
      }
    }
  }
`;

class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  };

  render() {
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{ id: this.props.id }}>
        {({ data, error, loading }) => {
          if (loading) return <p>Loading...</p>;
          if (error) return <ErrorMessage error={error} />;
          const { order } = data;
          if (!order) return <p>Not Found</p>;
          return (
            <OrderStyles data-test="order">
              <Head>
                <title>Sick Fits | View Order</title>
              </Head>
              <p>
                <span>Order ID:</span>
                <span>{order.id}</span>
              </p>
              <p>
                <span>Charge:</span>
                <span>{order.charge}</span>
              </p>
              <p>
                <span>Date:</span>
                <span>{format(order.createdAt, 'DD/MM/YYYY HH:MM ')}</span>
              </p>
              <p>
                <span>Total Item:</span>
                <span>{order.items.length}</span>
              </p>
              <p>
                <span>Total Amount:</span>
                <span>{formatMoney(order.total)}</span>
              </p>
              <div className="items">
                {order.items.map(item => (
                  <OrderItem item={item} key={item.id} />
                ))}
              </div>
            </OrderStyles>
          );
        }}
      </Query>
    );
  }
}

export default Order;
export { SINGLE_ORDER_QUERY };
