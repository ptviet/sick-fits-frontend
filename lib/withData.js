import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpointProd } from '../config';
import { LOCAL_STATE_QUERY } from '../components/Cart';

function createClient({ headers }) {
  return new ApolloClient({
    // uri: process.env.NODE_ENV === 'development' ? endpoint : endpointProd,
    uri: 'https://sp-sick-fits-yoga-prod.herokuapp.com',
    request: operation => {
      operation.setContext({
        fetchOptions: {
          credentials: 'include'
        },
        headers
      });
    },
    // Local data
    clientState: {
      resolvers: {
        Mutation: {
          toggleCart(_, variables, { cache }) {
            // Read the cartOpen value from the cache
            const { cartOpen } = cache.readQuery({ query: LOCAL_STATE_QUERY });
            // Write new cartOpen state
            const data = {
              data: {
                cartOpen: !cartOpen
              }
            };
            cache.writeData(data);
            return data;
          }
        }
      },
      defaults: {
        cartOpen: false
      }
    }
  });
}

export default withApollo(createClient);
