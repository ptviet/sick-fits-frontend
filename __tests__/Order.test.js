import { MockedProvider } from 'react-apollo/test-utils';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import wait from 'waait';
import Order, { SINGLE_ORDER_QUERY } from '../components/Order';
import { fakeOrder } from '../lib/testUtils';

const order = fakeOrder();

const mocks = [
  {
    request: { query: SINGLE_ORDER_QUERY, variables: { id: order.id } },
    result: {
      data: {
        order
      }
    }
  }
];

describe('<Order/>', () => {
  let wrapper;
  it('should render and match snapshot', async () => {
    wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Order id={order.id} />
      </MockedProvider>
    );
    await wait();
    wrapper.update();

    expect(toJson(wrapper.find('div[data-test="order"]'))).toMatchSnapshot();
  });
});
