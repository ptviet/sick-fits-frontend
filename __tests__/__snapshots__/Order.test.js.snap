// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order/> should render and match snapshot 1`] = `
<div
  className="OrderStyles-sc-4oqalm-0 camMij"
  data-test="order"
>
  <SideEffect(Head)>
    <Head />
  </SideEffect(Head)>
  <p>
    <span>
      Order ID:
    </span>
    <span>
      ord123
    </span>
  </p>
  <p>
    <span>
      Charge:
    </span>
    <span>
      ch_123
    </span>
  </p>
  <p>
    <span>
      Date:
    </span>
    <span>
      91/04/2018 10:04 
    </span>
  </p>
  <p>
    <span>
      Total Item:
    </span>
    <span>
      2
    </span>
  </p>
  <p>
    <span>
      Total Amount:
    </span>
    <span>
      $400
    </span>
  </p>
  <div
    className="items"
  >
    <OrderItem
      item={
        Object {
          "__typename": "OrderItem",
          "description": "modi tenetur amet modi reprehenderit omnis adipisci",
          "id": "3a430a73-c9e9-4bcc-b46a-41618965ffea",
          "image": "reprehenderit.jpg",
          "largeImage": "soluta.jpg",
          "price": 4234,
          "quantity": 1,
          "title": "non omnis consequatur enim quia autem et",
          Symbol(id): "OrderItem:3a430a73-c9e9-4bcc-b46a-41618965ffea",
        }
      }
      key="3a430a73-c9e9-4bcc-b46a-41618965ffea"
    >
      <div
        className="order-item"
      >
        <img
          alt="non omnis consequatur enim quia autem et"
          src="reprehenderit.jpg"
        />
        <div
          className="item-details"
        >
          <h2>
            non omnis consequatur enim quia autem et
          </h2>
          <p>
            Qty: 
            1
          </p>
          <p>
            Each: 
            $42.34
          </p>
          <p>
            Subtotal: 
            $42.34
          </p>
          <p>
            Description: 
            modi tenetur amet modi reprehenderit omnis adipisci
          </p>
        </div>
      </div>
    </OrderItem>
    <OrderItem
      item={
        Object {
          "__typename": "OrderItem",
          "description": "molestias assumenda quasi adipisci mollitia iste itaque",
          "id": "293e0904-82fd-403e-a08d-8d81098deec8",
          "image": "quia.jpg",
          "largeImage": "sed.jpg",
          "price": 4234,
          "quantity": 1,
          "title": "exercitationem omnis laborum exercitationem est sapiente laudantium",
          Symbol(id): "OrderItem:293e0904-82fd-403e-a08d-8d81098deec8",
        }
      }
      key="293e0904-82fd-403e-a08d-8d81098deec8"
    >
      <div
        className="order-item"
      >
        <img
          alt="exercitationem omnis laborum exercitationem est sapiente laudantium"
          src="quia.jpg"
        />
        <div
          className="item-details"
        >
          <h2>
            exercitationem omnis laborum exercitationem est sapiente laudantium
          </h2>
          <p>
            Qty: 
            1
          </p>
          <p>
            Each: 
            $42.34
          </p>
          <p>
            Subtotal: 
            $42.34
          </p>
          <p>
            Description: 
            molestias assumenda quasi adipisci mollitia iste itaque
          </p>
        </div>
      </div>
    </OrderItem>
  </div>
</div>
`;
